File: Components/WPF/DevExpress.SalesDemo.Wpf/VB/DevExpress.SalesDemo.Wpf/ViewModel/SectorsViewModel.vb
Imports DevExpress.Mvvm.POCO

Namespace DevExpress.SalesDemo.Wpf.ViewModel
    Public Class SectorsViewModel
        Inherits PageViewModel

        Public Shared Function Create() As SectorsViewModel
            Return ViewModelSource.Create(Function() New SectorsViewModel())
        End Function
        Protected Sub New()
        End Sub

        Public Overridable Property DailySalesBySectorViewModel() As PerformanceBarChartViewModel
        Public Overridable Property UnitSalesBySectorViewModel() As PerformanceBarChartViewModel
        Public Overridable Property PieGaugeRangeViewModel() As PieBarRangeViewModel
        Protected Overrides Sub Initialize()
            DailySalesBySectorViewModel = PerformanceBarChartViewModel.Create(ModuleType.Sectors, PerformanceViewMode.Daily)

            UnitSalesBySectorViewModel = PerformanceBarChartViewModel.Create(ModuleType.Sectors, PerformanceViewMode.Monthly)

            PieGaugeRangeViewModel = PieBarRangeViewModel.Create(ModuleType.Sectors)
        End Sub
    End Class
End Namespace

File: Components/WPF/DevExpress.SalesDemo.Wpf/CS/DevExpress.SalesDemo.Wpf/ViewModel/SectorsViewModel.cs
using DevExpress.Mvvm.POCO;

namespace DevExpress.SalesDemo.Wpf.ViewModel {
    public class SectorsViewModel : PageViewModel {
        public static SectorsViewModel Create() {
            return ViewModelSource.Create(() => new SectorsViewModel());
        }
        protected SectorsViewModel() { }

        public virtual PerformanceBarChartViewModel DailySalesBySectorViewModel { get; protected set; }
        public virtual PerformanceBarChartViewModel UnitSalesBySectorViewModel { get; protected set; }
        public virtual PieBarRangeViewModel PieGaugeRangeViewModel { get; protected set; }
        protected override void Initialize() {
            DailySalesBySectorViewModel = PerformanceBarChartViewModel.Create(ModuleType.Sectors, PerformanceViewMode.Daily); ;
            UnitSalesBySectorViewModel = PerformanceBarChartViewModel.Create(ModuleType.Sectors, PerformanceViewMode.Monthly); ;
            PieGaugeRangeViewModel = PieBarRangeViewModel.Create(ModuleType.Sectors);
        }
    }
}
